### ISSUE TITLE
[Feature Request] Fetch & Track Indexer Stats (Upload / Download Ratio, Points, etc.)

### ISSUE THREAD

**Issue by some-guy-23** [Prowlarr/Prowlarr#1203]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

I couldn't find whether this has been discussed before, but I think it would open up a lot of possibilities if Prowlarr could also fetch and track information from your indexers. For example the amount you've upload / downloaded, and any "bonus points" you've accumulated. This could then be used for various automated functionality.

### Describe the solution you'd like

For an indexer definition, there could be an optional method that defines how to fetch various account information like Upload / Download and "Bonus Points". This would only be fetched once a day or such to ensure there's not extra load on indexers (not user configurable).

You could then act on this information gathered, such as disabling an indexer from being used when the ratio is below a certain amount, or sending a notification to the user. You could even execute actions, such as redeeming bonus points for upload credit, etc.

### Describe alternatives you've considered

Basically what I'm already doing today, which is doing this manually.

### Anything else?

This would be a huge feature and change I know, but at the same time I think it could be a pretty huge differentiator vs other software that fills a similar role as Prowlarr. There would be a huge amount to figure out, but today I'm mostly wondering if this would even be considered?

[AB#4101](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/4101)

**Comment by techguruz** [comment]
this is something i would like to see having used pt-plugin-pro in the past.

**Comment by rpatterson** [comment]
I'm very interested in this. I might even learn the .net stack if I was confident such a contribution was wanted and would be supported through to merge. I'm assuming Prowlarr is already scraping tracker sites for trackers that don't offer RSS feeds, so it would be a shame to duplicate all that logic (auth, cookies, user-agent, other headers, Cloud Flare, rate limits, etc.) in another tool. Unfortunately, it looks like not many other folks are interested in this, so I doubt the maintainers would be eager to take on the maintenance of seldom used code paths.

Another piece of information it would be useful to have some sort of generalized API to access would be the per-release stats, particularly seeding time to avoid hit and runs. The local download client may only offer a started timestamp and a completed timestamp for each item and subtracting the former from the latter may not represent seeding time for various reasons: downtime, paused items, etc.. Even download clients that track cumulative time spent seeding, it still might be different from what the tracker thinks. Not having the tracker's sense of time spent seeding puts any attempt to automate when it's ok to remove upgraded releases at risk of incurring HnRs.

**Comment by techguruz** [comment]
> I'm very interested in this. I might even learn the .net stack if I was confident such a contribution was wanted and would be supported through to merge. I'm assuming Prowlarr is already scraping tracker sites for trackers that don't offer RSS feeds, so it would be a shame to duplicate all that logic (auth, cookies, user-agent, other headers, Cloud Flare, rate limits, etc.) in another tool. Unfortunately, it looks like not many other folks are interested in this, so I doubt the maintainers would be eager to take on the maintenance of seldom used code paths.
> 
> Another piece of information it would be useful to have some sort of generalized API to access would be the per-release stats, particularly seeding time to avoid hit and runs. The local download client may only offer a started timestamp and a completed timestamp for each item and subtracting the former from the latter may not represent seeding time for various reasons: downtime, paused items, etc.. Even download clients that track cumulative time spent seeding, it still might be different from what the tracker thinks. Not having the tracker's sense of time spent seeding puts any attempt to automate when it's ok to remove upgraded releases at risk of incurring HnRs.

I think people just don't know about it. Maybe implementing it as a custom script much like the arr scripts from linuxserver [https://github.com/RandomNinjaAtk/arr-scripts?tab=readme-ov-file](url) or a docker mod [https://github.com/linuxserver/docker-mods](url) could be worth reaching out to the guys over at linuxserver and see if they could offer any help. To be honest though this seems like it would be a whole new arr webapp on its own unless it could be officially implemented into prowlarr v2 or been v3.

**Comment by rpatterson** [comment]
> Maybe implementing it as a custom script

I might be misunderstanding your suggestion, but I don't think that addresses the issue of adding these features to Prowlarr or implementing them separately and duplicating much of Prowlarr's work. Whether deployed as a custom script, a docker mod, or a separate service, it's still scraping tracker web UIs, it's still duplicating Prowlarr's work.

> To be honest though this seems like it would be a whole new arr webapp on its own unless it could be officially implemented into prowlarr v2 or been v3.

Well, if not added to Prowlarr, then yes it would at least be it's own library and would end up duplicating much of Prowlarr's scraping logic:

> I'm assuming Prowlarr is already scraping tracker sites for trackers that don't offer RSS feeds, so it would be a shame to duplicate all that logic (auth, cookies, user-agent, other headers, Cloud Flare, rate limits, etc.) in another tool.
